README.md

```diff
@@ -1,35 +1,466 @@
+
+  ```
+  using Swrve;
+
+  public class CustomPushNotificationListener : ISwrvePushNotificationListener {
+    public void OnRemoteNotification(RemoteNotification notification) {
+      // CUSTOM CODE HERE
+    }
+  }
+  SwrveComponent.Instance.SDK.PushNotificationListener = new CustomPushNotificationListener();
+  ```
+
+### Creating and uploading your iOS Cert ###
+
+To enable your app to send push notifications to Unity iOS devices, you require a push certificate. A push certificate authorizes an app to receive push notifications and authorizes a service to send push notifications to an app. For more information, see [How Do I Manage iOS Push Certificates?](http://docs.swrve.com/faqs/push-notifications/manage-ios-push-certificates-for-push-notifications/)
+
+### Provisioning your app ###
+
+Each time your app is published for distribution, you must provision it against the push certificate that has been created for the app.
+
+Android Push Notifications
+-
+
+Unity Android uses a native Android plugin (provided as part of the Unity SDK), so you must include this plugin in your project as detailed below.
+
+To integrate Swrve’s push notification functionality for Unity Android, you must be using Unity SDK version 2.9 or later and Unity Android Pro 4.0.0+.
+
+1. Copy the JAR files located under `plugins/SwrveGooglePlayPushNotification` into `Assets/Plugins/Android`.
+
+2. Copy the `AndroidManifest.xml` file (in `plugins/SwrveGooglePlayPushNotification`) into `Assets/Plugins/Android` and replace the sample package name `com.example.gcm` with the package name for your app.
+
+  If you are already providing a custom `AndroidManifest.xml` or your project generates a different XML from the sample one, you can make a copy from the Staging folder once you make your first Android player build from Unity. Then include the required permissions, BroadcastListener, service and custom activities.
+
+3. Initializing the SDK. You must provide the project number obtained from the Google Developer Console to the SDK on initialization.
+
+  ```
+  #if UNITY_ANDROID
+  SwrveComponent.Instance.Config.PushNotificationEnabled = true;
+  SwrveComponent.Instance.Config.GCMSenderId = SENDER_ID;
+  SwrveComponent.Instance.Config.GCMPushNotificationTitle = "Awesome app";
+  #endif // UNITY_ANDROID
+  ```
+
+  Changes to `SwrveComponent.Instance.Config` must occur before calling `SwrveComponent.Instance.Init`.
+
+### Using a custom `GcmReceiver` ###
+
+If you are already using push notifications, follow the steps below to integrate Swrve’s push notification functionality:
+
+1. Copy `swrvesdkpushsupport.jar` (located under `plugins/SwrveGooglePlayPushNotification`) into `Assets/Plugins/Android`.
+
+2. Add the following lines to the zzd method of your custom `GcmReceiver`:
+
+  ```
+  @Override
+  public void zzd(Context context, Intent intent) {
```

Comment:
A better approach would be to have a service that calls ours if it can find the _p and not touch the zzd/etc methods that are hidden and seem to keep changing name with Play versions